tests/python/pants_test/backend/jvm/tasks/test_coursier_resolve.py

```diff
@@ -59,6 +60,30 @@ def test_resolve_specific(self):
     compile_classpath = self.resolve([jar_lib, scala_lib])
     self.assertEquals(1, len(compile_classpath.get_for_target(jar_lib)))
     self.assertEquals(0, len(compile_classpath.get_for_target(scala_lib)))
+    
+  def test_resolve_with_remote_url(self):
+    dep_with_url = JarDependency('a', 'b', 'c',
+                                 url='http://central.maven.org/maven2/junit/junit/4.12/junit-4.12.jar')
+    dep_with_url_lib = self.make_target('//:a', JarLibrary, jars=[dep_with_url])
+    
+    compile_classpath = self.resolve([dep_with_url_lib])
+    # Get paths on compile classpath and assert that it starts with '.../coursier/cache/relative'
+    paths = [tup[1] for tup in compile_classpath.get_for_target(dep_with_url_lib)]
+    self.assertTrue(any('.pants.d/coursier/cache/relative/' in path for path in paths))
+  
+  def test_resolve_with_local_url(self):
+    file_d, tmp_path = tempfile.mkstemp(suffix='.jar')
```

Comment:
should be able to use https://github.com/pantsbuild/pants/blob/4c21f01a913d569db5b160a719c1805588d931ff/src/python/pants/util/contextutil.py#L156
so cleanup is automatically done.
```
with temporary_file(suffix='.jar') as f:
  ...
```